### PR TITLE
Leaving network interface and resolved configuration to the system

### PR DESCRIPTION
**Pull request by Yayg**
Hi,

First thanks for maintaining this package which is really helpful for custom installation.
I have a proposal here because I ran into the same problems as people in issues:
- https://github.com/home-assistant/supervised-installer/issues/339
- https://github.com/home-assistant/supervised-installer/issues/360
An also into issues with the network interface file which was overwriting my lacp configuration for instance.

Thus the proposal is to leave network interface and resolved configuration to the system administrator.
Now that being said most of this configuration is handled by Debian installer so it should not be a problem for anyone.

Sincerely

### CODE REVIEW

**Comment by agners** [comment]
We do mention that systemd-resolved must be installed in the README, so this commands should work in general. Also, systemd-resolved is mandatory for Supervisor, as the DNS plug-in uses it to resolve things via mDNS for the Core/Supervisor/Add-ons etc.

But if we can get rid of unnecessary interference, I am not against it. Is installing systemd-resolved automatically enabling it on Debian 12, so that manually enabling it no longer necessary?

**Comment by Yayg** [comment]
> We do mention that systemd-resolved must be installed in the README, so this commands should work in general. Also, systemd-resolved is mandatory for Supervisor, as the DNS plug-in uses it to resolve things via mDNS for the Core/Supervisor/Add-ons etc.

Systemd-resolved is installed and the configuration should not be directly changed because it will conflict with the maintainer's default. Also it creates situations where the URL becomes unresolved because the configuration given is not correct ATM.

> But if we can get rid of unnecessary interference, I am not against it. Is installing systemd-resolved automatically enabling it on Debian 12, so that manually enabling it no longer necessary? 

Yes the necessary is already there in the base package. Also, as you mentioned, it is explicitly said in the supervised guide to install the dependencies, so I guess leave it all to the system is the best way to go.

Same for interfaces config, they broke the network for me.

**Comment by ikifar2012** [comment]
While I do agree the NetworkManager config has caused issues in some scenarios and I am OK with looking into removing that, I do not think removing `systemd-resolved` is necessary

**Comment by Yayg** [comment]
> While I do agree the NetworkManager config has caused issues in some scenarios and I am looking into possible solutions to that, I do not think removing `systemd-resolved` is necessary 

As seen and detailed in https://github.com/home-assistant/supervised-installer/issues/339 systemd-resolved is problematic with the same consequences.

Depending on the network setup the user may have to specify manually in the configuration it's internal DNS.

Moreover, any reinstallation of the package wipes the configuration handwritten if necessary.

**Comment by Yayg** [comment]
Btw, if any configuration should be added by HA or it should be in `etc/systemd/resolved.conf.d/`. Not directly in the main configuration file.

**Comment by litinoveweedle** [comment]
> Depending on the network setup the user may have to specify manually in the configuration it's internal DNS.
> 
> Moreover, any reinstallation of the package wipes the configuration handwritten if necessary.

I can confirm this, I reported also in #374. During each package upgrade, install or re-install my DNS setting are lost and manual recovery is required. At least now I already know what to do, as for few first times I had to go through systemd DNS implementation rabbit hole. Again my understanding is, that supervised is for skilled admin users, who probably already have working system setup including networking and can maintain it (if the decide even without NetworkManager).

File: homeassistant-supervised/etc/systemd/resolved.conf [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-# Defaults can be restored by simply deleting this file.
-#
-# See resolved.conf(5) for details
-
-[Resolve]
-#DNS=
-#FallbackDNS=1.1.1.1 8.8.8.8 1.0.0.1 8.8.4.4 2606:4700:4700::1111 2001:4860:4860::8888 2606:4700:4700::1001 2001:4860:4860::8844
-#Domains=
-DNSSEC=no
-DNSOverTLS=no
```
Comment (agners) [inline homeassistant-supervised/etc/systemd/resolved.conf:19]: These two configs got most likely copied over from the HAOS config at https://github.com/home-assistant/operating-system/commit/e3120df5d08da5aade13e5bcbf9a094cd538fde3.

Buildroot (used to build HAOS) by default sets `DNSOverTLS` to opportunistic mode, which did cause issues in certain cases. Hence we opt out of it. Debian uses the systemd default configuration which is off.

For DNSSEC, Debian explicitly turns it off by default.

So it is fairly safe to not set these two options, as their default is `no` already.

File: homeassistant-supervised/etc/systemd/resolved.conf [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-#MulticastDNS=yes
-#LLMNR=yes
-#Cache=yes
-DNSStubListener=no
```
Comment (agners) [inline homeassistant-supervised/etc/systemd/resolved.conf:23]: This setting most likely came from HAOS too, particularly this commit: https://github.com/home-assistant/operating-system/commit/7c25f7c18723841d7b558543612cab18c93d3910

To be safe, we should keep this config by default, but as @Yayg mentions, in a file at etc/systemd/resolved.conf.d/.

**Review by agners — COMMENTED** [review]
> While I do agree the NetworkManager config has caused issues in some scenarios and I am looking into possible solutions to that, I do not think removing systemd-resolved is necessary

From what I can see, this PR does not remove the dependency to systemd-resolved (no changes to the control file). And this is good, since we use systemd-resolved to do mDNS resolution in the Supervisor (add-ons/Core) ecosystem via CoreDNS plug-in. So systemd-resolved is indeed a hard dependency of the Supervised installation method.

From my standpoint it makes sense to follow Debian best practice and not overwrite or mess otherwise with other packages. From what I can tell, at least today, the systemd-resolved package should enable systemd-resolved by default (at least this is what the package description says: "Installing this package automatically overwrites /etc/resolv.conf and switches it to be managed by systemd-resolved.").

**Pull request closed without merging**